lib/commands/web.js

```diff
@@ -0,0 +1,193 @@
+import { retryInterval } from 'asyncbox';
+import _ from 'lodash';
+import { errors } from 'mobile-json-wire-protocol';
+import logger from '../logger';
+
+
+let commands = {}, helpers = {}, extensions = {};
+
+commands.setUrl = async function (url) {
+  logger.debug(`Attempting to set url '${url}'`);
+  if (this.isWebContext()) {
+    // make sure to clear out any leftover web frames
+    this.curWebFrames = [];
+    this.processingRemoteCmd = true;
+    await this.remote.navToUrl(url);
+    this.processingRemoteCmd = false;
+  } else {
+    // in the future, detect whether we have a UIWebView that we can use to
+    // make sense of this command. For now, and otherwise, it's a no-op
+    throw new errors.NotImplementedError();
+  }
+};
+
+commands.getUrl = async function () {
+  if (this.isWebContext()) {
+    this.processingRemoteCmd = true;
+    let value = await this.remote.execute('window.location.href');
+    this.processingRemoteCmd = false;
+    return value;
+  } else {
+    throw new errors.NotImplementedError();
+  }
+};
+
+commands.title = async function () {
+  if (this.isWebContext()) {
+    let res =  await this.executeAtom('title', []);
+    return res.value;
+  } else {
+    throw new errors.NotImplementedError();
+  }
+};
+
+commands.findWebElementOrElements = async function (strategy, selector, ctx, many) {
+  let atomsElement = this.getAtomsElement(ctx);
+  let res;
+  let doFind = async () => {
+    res = await this.executeAtom(`find_element${many ? 's' : ''}`, [strategy, selector, atomsElement]);
+    return res.status === 0 && res.value !== null;
+  };
+  await this.implicitWaitForCondition(doFind);
+  return res.value;
+};
+
+commands.click = async function (elementId) {
```

Comment:
This is going to conflict with the gesture one. This probably need to be a `webClick` helper.